File: patterns/single/hueStripe.py
import random
import pattern
from utils.color_utils import hsvToRGB
from math import cos

from patterns.single import registerPattern 

class HueStripe(pattern.Pattern):
	def __init__(self, channel=0):
		pattern.Pattern.__init__(self, channel)

		self.hue = random.randint(0, 359);
		self.speed = float(random.randint(10, 60))
		self.direction = random.randint(0, 2)

		self.range = 40
		self.swing = self.range / 2.0

		self.offset = 0;

	def update(self):
		self.offset += self.speed / 100.0

	def shader(self, coords, led_num):
		dir_var = coords[self.direction]
		
		hue = (self.hue + ((cos(self.offset * 0.125 + dir_var) * self.swing) + self.range / 2)) % 360

		return map(lambda x: x * 255 % 256, hsvToRGB(hue, 1, 1))

registerPattern(HueStripe, "Hue Stripe")

File: patterns/group/hueStripe.js
var util = require('util');
var Pattern = require(patternRoot + 'single/hueStripe');
var colorUtils = require(appRoot + 'libs/color_utils');

var Stripe = function()
{
	Stripe.super_.call(this);

	this.channel_offset = 1000;
}

util.inherits(Stripe, Pattern);

Stripe.prototype.init = function()
{
	this.offset = this.channel * this.channel_offset;
}

Stripe.prototype.shader = function(coords, led_num, tent_coords, lantern_num)
{
	var overall_coords = [
        coords.point[0] + tent_coords.point[0],
	    coords.point[1] + tent_coords.point[1],
	    coords.point[2] + tent_coords.point[2]
    ];

	var dir_var = overall_coords[this.direction];

	var remappedHue = colorUtils.remap(this.hue, 0, 1, 0, 255);
	var bigHue = (remappedHue + ((Math.cos(this.offset * 0.125 + (1.0 + dir_var)) * this.swing) + this.range / 2)) % 255; 
	var smallHue = colorUtils.remap(bigHue, 0, 255, 0, 1);
	return colorUtils.hsv(smallHue, 1, 1);
}

module.exports = Stripe;
